e2e/test/helpers/templates/PoolingOverAmqp.cs

```diff
@@ -78,13 +79,10 @@ public static class PoolingOverAmqp
 
                     if (initOperation != null)
                     {
-                        operations.Add(initOperation(deviceClient, testDevice, testDeviceCallbackHandler));
+                        await initOperation(deviceClient, testDevice, testDeviceCallbackHandler).ConfigureAwait(false);
                     }
                 }
 
-                await Task.WhenAll(operations).ConfigureAwait(false);
```

Comment:
No need to this this list of tasks for the init operation when we can await it directly.